conda/plugins/manager.py

```diff
@@ -361,6 +378,33 @@ def invoke_post_solves(
         for hook in self.get_hook_results("post_solves"):
             hook.action(repodata_fn, unlink_precs, link_precs)
 
+    def load_configuration_parameters(self) -> None:
+        """
+        Iterates through all registered configuration parameters and adds them to the context class.
+
+        :raises PluginError: Happens when plugins attempt to override existing parameters
+
+        TODO:
+            - Need to add configuration parameters to ``Context.category_map`` and
+              ``Context.description_map``
+        """
+        existing_parameters = context.list_parameters()
+        error_params = []
+
+        for name, loader in self.get_configuration_parameters().items():
+            if name in existing_parameters:
+                error_params.append(name)
+            name = loader._set_name(name)
+            setattr(Context, name, loader)
+            setattr(Context, "parameter_names", Context.parameter_names + (name,))
```

Comment:
This is a bit of code smell that we need to overcome with a context API, it's not great we have to monkeypatch `Context` like this.

Reply:
E.g. in the Context class (not sure if this exists already)
```python
def add_parameter(self, name, loader):
    setattr(self, name, loader)
    self.parameter_names += (name,)
```

But maybe there is more to do in the way parameter names are handled, that would not require setting instance (or class) attributes like this

Reply:
I agree that there needs to be an API on Context for this.

I think this might have to be a `@classmethod` though. I remember it not working when I tried to set it on the instance itself.